src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -337,8 +337,79 @@ def _check_registry_health(cmd, registry_name, ignore_errors):
+
+    registry = None
+
+    # retrieve registry
+    try:
+        registry, _ = get_registry_by_name(cmd.cli_ctx, registry_name)
+    except CLIError:
+        logger.warning('Registry resource of "%s" must be accessible to verify DNS routings of its private endpoints.',
+                       registry_name)
+        return
+
+    if not registry.private_endpoint_connections:
+        logger.warning('Registry "%s" doesn\'t have private endpoints to verify DNS routings.', registry_name)
+        return
+
+    if not is_valid_resource_id(vnet_of_private_endpoint):
+        res = parse_resource_id(registry.id)
+        vnet_of_private_endpoint = resource_id(name=vnet_of_private_endpoint, resource_group=res['resource_group'],
+                                               namespace='Microsoft.Network', type='virtualNetworks',
+                                               subscription=res['subscription'])
+
+    # retrieve FQDNs for registry and its data endpoint
+    pe_ids = [e.private_endpoint.id for e in registry.private_endpoint_connections if e.private_endpoint]
+    dns_mappings = {}
+    for pe_id in pe_ids:
+        res = parse_resource_id(pe_id)
+        network_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_NETWORK,
+                                                 subscription_id=res['subscription'])
+        pe = network_client.private_endpoints.get(res['resource_group'], res['name'])
+        if bool(re.match(vnet_of_private_endpoint, pe.subnet.id, re.I)):
+            nic_id = pe.network_interfaces[0].id
+            nic_res = parse_resource_id(nic_id)
+            nic = network_client.network_interfaces.get(nic_res['resource_group'], nic_res['name'])
+            for dns_config in nic.ip_configurations:
+                if dns_config.private_link_connection_properties.fqdns[0] in dns_mappings:
+                    err = ('Registry "%s" has more than one private endpoint in the vnet of "%s".'
+                           ' DNS routing will be unreliable')
+                    logger.warning(err, registry_name, vnet_of_private_endpoint)
+                dns_mappings[dns_config.private_link_connection_properties.fqdns[0]] = dns_config.private_ip_address
+
+    dns_ok = True
+    if not dns_mappings:
+        logger.warning('Registry "%s" doesn\'t have private endpoints in the vnet of "%s".',
+                       registry_name, vnet_of_private_endpoint)
+        return
+
+    for fqdn in dns_mappings:
+        try:
+            result = socket.gethostbyname(fqdn)
+            if result != dns_mappings[fqdn]:
+                err = 'DNS routing to registry "%s" through private IP is incorrect. Expect: %s, Actual: %s'
+                logger.warning(err, registry_name, dns_mappings[fqdn], result)
+                dns_ok = False
+        except Exception as e:  # pylint: disable=broad-except
+            logger.warning('Error resolving DNS for %s. Ex: %s', fqdn, e)
+            dns_ok = False
+
+    if dns_ok:
```

Comment:
Question only: Do you see any value in printing the DNS mappings (for both ok and non-ok scenarios)? 
Do you think It might help customers with self-diagnosis?

```
registry.azurecr.io             10.0.0.1
registry.eastus.data.azurecr.io 10.0.0.2
registry.westus.data.azurecr.io 10.0.0.3
```

Reply:
Right now we only have stderr for non structured info(stdout is for json data), I suggest keep it terse so error can be spotted easily. 
However if feedback shows users don't get the error easily, we can incorporate errors in the pretty formating
```
registry.azurecr.io                        10.0.0.1
registry.eastus.data.azurecr.io            10.0.0.2
registry.westus.data.azurecr.io            10.0.0.3
registry.westeurope.data.azurecr.io    (dns routing error)
```